google/cloud/security/notifier/notifier.py

```diff
@@ -0,0 +1,142 @@
+
+# Format: flags.DEFINE_<type>(flag_name, default_value, help_text)
+# Example:
+# https://github.com/google/python-gflags/blob/master/examples/validator.py
+flags.DEFINE_string('timestamp', None, 'Snapshot timestamp')
+flags.DEFINE_string('pipeline', None, 'Pipeline to use')
+flags.DEFINE_string('config', None, 'Config file to use', short_name='c')
+
+LOGGER = log_util.get_logger(__name__)
+OUTPUT_TIMESTAMP_FMT = '%Y%m%dT%H%M%SZ'
+
+def find_pipelines(pipeline_name):
+    """Get the first class in the given sub module
+
+    Return:
+        The class in the sub module
+    """
+    try:
+        module = importlib.import_module(
+            'google.cloud.security.notifier.pipelines.{0}'.format(
+                pipeline_name))
+        for filename in dir(module):
+            obj = getattr(module, filename)
+
+            if inspect.isclass(obj) and issubclass(obj, NotificationPipeline) \
+               and obj is not NotificationPipeline:
+                return obj
+    except ImportError:
+        return None
+
+def _get_timestamp(statuses=('SUCCESS', 'PARTIAL_SUCCESS')):
+    """Get latest snapshot timestamp.
+
+    Returns:
+        The latest snapshot timestamp string.
+    """
+
+    latest_timestamp = None
+    try:
+        latest_timestamp = dao.Dao().get_latest_snapshot_timestamp(statuses)
+    except db_errors.MySQLError as err:
+        LOGGER.error('Error getting latest snapshot timestamp: %s', err)
+
+    return latest_timestamp
+
+def main(_):
+    """main function"""
+    timestamp = FLAGS.timestamp if FLAGS.timestamp is not None \
+                else _get_timestamp()
+
+    if FLAGS.config is None:
+        LOGGER.error('You must specify a notification pipeline')
+        exit()
+
+    if FLAGS.pipeline is None:
+        LOGGER.error('You must specify a notification pipeline')
+        exit()
+
+    notifier_configs = FLAGS.FlagValuesDict()
+    configs = file_loader.read_and_parse_file(FLAGS.config)
+
+    # get violations
+    v_dao = violation_dao.ViolationDao()
+    violations = {
+        'violations': v_dao.get_all_violations(
+                      timestamp,
+                      'violations'),
+        'bucket_acl_violations': v_dao.get_all_violations(
+                      timestamp,
+                      'buckets_acl_violations')
+    }
+    for retrieved_v in violations:
+        LOGGER.info('retrieved %d violations for resource \'%s\'' % (
+            len(violations[retrieved_v]), retrieved_v))
+
+    # build notification pipelines
+    pipelines = []
+    for resource in configs:
+        if violations.get(resource['resource']) is None:
+            LOGGER.error('The resource name \'%s\' is invalid, skipping' %
+                         resource['resource'])
+            continue
+        if resource['should_notify'] is False:
+            continue
+        for pipeline in resource['pipelines']:
+            print pipeline
+            chosen_pipeline = find_pipelines(pipeline['name'])
```

Comment:
I'm not sure, because right now I'm also dynamically loading the pipeline class from the module as well.

```
        module = importlib.import_module(
            'google.cloud.security.notifier.pipelines.{0}'.format(
                pipeline_name))
        for filename in dir(module):
            obj = getattr(module, filename)

            if inspect.isclass(obj) and issubclass(obj, NotificationPipeline) \
               and obj is not NotificationPipeline:
                return obj
```